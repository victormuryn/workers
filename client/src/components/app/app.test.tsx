import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';

import renderer, {create, act} from 'react-test-renderer';

import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

import App from './app';

const mockStore = configureMockStore();

describe(`App tests`, () => {
  it(`App not authenticated renders corrects`, () => {
    const store = mockStore({
      user: {
        isAuthenticated: false,
        accountType: null,
      },
    });

    let tree: renderer.ReactTestRenderer;
    act(() => {
      tree = create(
        <Provider store={store}>
          <Router>
            <App />
          </Router>
        </Provider>,
      );
    });

    // @ts-ignore
    const json = tree.toJSON();
    expect(json).toMatchSnapshot();
  });

  it(`App authenticated renders corrects`, () => {
    const store = mockStore({
      user: {
        isAuthenticated: true,
        accountType: `client`,
        username: `hellcaster`,
      },
    });

    let tree: renderer.ReactTestRenderer;
    act(() => {
      tree = create(
        <Provider store={store}>
          <Router>
            <App />
          </Router>
        </Provider>,
      );
    });

    // @ts-ignore
    const json = tree.toJSON();
    expect(json).toMatchSnapshot();
  });
});
