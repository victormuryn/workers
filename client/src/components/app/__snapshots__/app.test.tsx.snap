// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests App authenticated renders corrects 1`] = `
Array [
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <a
        className="inner-header__logo navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Workers
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="ms-auto me-0 navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            aria-current={null}
            className="ms-2 nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/projects"
            onClick={[Function]}
          >
            Проєкти
          </a>
          <a
            className="ms-2 nav-link"
            data-rb-event-key="#features"
            href="#features"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Features
          </a>
          <div
            className="ms-2 dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              data-rb-event-key={null}
              href="#"
              id="header-profile"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              hellcaster
            </a>
          </div>
          <a
            className="ms-md-5 btn btn-success"
            disabled={false}
            href="/create"
            onClick={[Function]}
          >
            Створити проєкт
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <footer
    className="footer container-fluid"
    style={
      Object {
        "background": "linear-gradient(
      179deg,
      transparent 0,
      transparent 45px,
      #fff 45px,
      #fff 100%
    )",
      }
    }
  >
    <div
      className="footer__wrapper row row-cols-md-3 row-cols-1"
    >
      <div
        className="footer__column col-md-12 order-md-last"
      >
        <h2
          className="footer__logo"
        >
          <a
            href="/"
          >
            Workers
          </a>
        </h2>
        <p
          className="footer__copy"
        >
          © 2021 workers.com.ua — всі права захищені
        </p>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Реклама на сайті
        </a>
        <a
          href="#"
        >
          Зв'язатися з адміністрацією
        </a>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Угода користувача
        </a>
        <a
          href="#"
        >
          Допомога
        </a>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Замовники
        </a>
        <a
          href="#"
        >
          Виконавці
        </a>
        <a
          href="#"
        >
          Блог
        </a>
        <a
          href="#"
        >
          Безпечна угода
        </a>
      </div>
    </div>
  </footer>,
]
`;

exports[`App tests App not authenticated renders corrects 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__line"
      >
        <h2
          className="header__logo"
        >
          <a
            href="/"
          >
            Workers
          </a>
        </h2>
        <div
          className="header__btns-wrapper"
        >
          <a
            className="header__btn btn btn-success"
            disabled={false}
            href="/auth"
            onClick={[Function]}
          >
            Реєстрація
          </a>
          <a
            className="header__btn btn btn-primary"
            disabled={false}
            href="/login"
            onClick={[Function]}
          >
            Увійти
          </a>
        </div>
      </div>
      <div
        className="header__main-text"
      >
        <h1
          className="header__title"
        >
          Шукаєш 
          <span>
            спеціаліста
          </span>
          ?
        </h1>
        <div
          className="header__btns-wrapper"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="header__main-btn
                header__btn header__main-btn--get-master btn btn-success"
              disabled={false}
              type="button"
            >
              Знайти спеціаліста
            </button>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="header__main-btn
                header__btn header__main-btn--get-work btn btn-success"
              disabled={false}
              type="button"
            >
              Знайти роботу
            </button>
          </a>
        </div>
      </div>
    </div>
  </header>,
  <section
    className="why text-center container"
  >
    <h2
      className="why__title"
    >
      Чому саме ми?
    </h2>
    <div
      className="row"
    >
      <div
        className="col-md-4 col-12"
      >
        <div
          className="why__item-img-wrapper"
        >
          <img
            alt="№1 у світі"
            className="why__item-img"
            src="./img/one.svg"
          />
        </div>
        <h5
          className="why__item-title"
        >
          Унікальний сервіс
        </h5>
        <p
          className="why__item-text"
        >
          На сьогодні наш сервіс має найбільшу базу клієнтів та немає аналогів у всьому світі, тому тобі швидко вдасться знайти потрібного фахівця!
        </p>
      </div>
      <div
        className="col-md-4 col-12"
      >
        <div
          className="why__item-img-wrapper"
        >
          <img
            alt="Перевірений професіонал"
            className="why__item-img"
            src="./img/worker.svg"
          />
        </div>
        <h5
          className="why__item-title"
        >
          Величезна кількість робітників
        </h5>
        <p
          className="why__item-text"
        >
          Понад 1000 професіоналів готові допомогти тобі у будь-який момент. Усі спеціалісти проходять обов'язкову перевірку, щоб ти був впевнений у якості роботи.
        </p>
      </div>
      <div
        className="col-md-4 col-12"
      >
        <div
          className="why__item-img-wrapper"
        >
          <img
            alt="Захист"
            className="why__item-img"
            src="./img/plane.svg"
          />
        </div>
        <h5
          className="why__item-title"
        >
          100% надійно
        </h5>
        <p
          className="why__item-text"
        >
          Перед початком роботи гроші резервуються, щоб ти був переконаний в успішному завершенні проєкту. У разі, якщо робота не буде виконана — ми вернемо гроші!
        </p>
      </div>
    </div>
  </section>,
  <section
    className="custom wow fadeIn text-center container"
  >
    <h2
      className="custom__title"
    >
      Працюємо так, як зручно тобі
    </h2>
    <picture>
      <source
        srcSet="./img/work.webp, ./img/work@2x.webp 2x"
        type="image/webp"
      />
      <source
        srcSet="./img/work.jpg, ./img/work@2x.jpg 2x"
        type="image/jpeg"
      />
      <img
        alt="Стараємося для Вас"
        className="custom__img"
        src="./img/work.jpg"
      />
    </picture>
    <p
      className="custom__text"
    >
      Резервація грошей, гнучка система роботи, простота, швидка технічна підтримка: усе, що потрібно для комфортної співпраці!
    </p>
  </section>,
  <section
    className="how-it-works container-fluid"
  >
    <h2
      className="how-it-works__title"
    >
      Робота у 5 простих кроків
    </h2>
    <ol
      className="how-it-works__list"
    >
      <li
        className="how-it-works__item how-it-works__item--create
             wow slideInLeft"
      >
        <h4
          className="how-it-works__item-title"
        >
          1 Створіть проєкт
        </h4>
        <p
          className="how-it-works__item-text"
        >
          Детально опишіть завдання та побажання для виконавців
        </p>
      </li>
      <li
        className="how-it-works__item how-it-works__item--choose
             wow slideInRight"
      >
        <h4
          className="how-it-works__item-title"
        >
          2 Виберіть виконавця
        </h4>
        <p
          className="how-it-works__item-text"
        >
          Оберіть виконавця за ціною, термінами, відгуками та портфоліо
        </p>
      </li>
      <li
        className="how-it-works__item how-it-works__item--reserve
             wow slideInLeft"
      >
        <h4
          className="how-it-works__item-title"
        >
          3 Зарезервуйте кошти
        </h4>
        <p
          className="how-it-works__item-text"
        >
          Після вибору, зарезервуйте кошти та розпочніть роботу з виконавцем
        </p>
      </li>
      <li
        className="how-it-works__item how-it-works__item--follow
            wow slideInRight"
      >
        <h4
          className="how-it-works__item-title"
        >
          4 Слідкуйте за ходом роботи
        </h4>
        <p
          className="how-it-works__item-text"
        >
          Відстежуйте виконання роботи, контролюйте деталі у чаті з виконавцем
        </p>
      </li>
      <li
        className="how-it-works__item how-it-works__item--reviews
             wow slideInLeft"
      >
        <h4
          className="how-it-works__item-title"
        >
          5 Завершіть виконаний проєкт
        </h4>
        <p
          className="how-it-works__item-text"
        >
          Після закінчення роботи підтвердіть виконання та обміняйтеся відгуками
        </p>
      </li>
      <li
        className="how-it-works__item how-it-works__item--done
             wow slideInRight"
      >
        <div
          className="visually-hidden"
        >
          Кінець
        </div>
      </li>
    </ol>
  </section>,
  <section
    className="categories wow fadeIn container"
  >
    <h2
      className="categories__title"
    >
      Найпопулярніші категорії
    </h2>
    <div
      className="row row-cols-lg-5"
    >
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Сантехніка"
            className="categories__item-img"
            src="./img/pipeline.svg"
          />
          <h5
            className="categories__item-title"
          >
             Сантехніка
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Електрика"
            className="categories__item-img"
            src="./img/plug.svg"
          />
          <h5
            className="categories__item-title"
          >
            Електрика
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Доставка"
            className="categories__item-img"
            src="./img/delivery.svg"
          />
          <h5
            className="categories__item-title"
          >
            Доставка
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Автосервіс"
            className="categories__item-img"
            src="./img/service.svg"
          />
          <h5
            className="categories__item-title"
          >
            Автосервіс
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Ремонт"
            className="categories__item-img"
            src="./img/paint-bucket.svg"
          />
          <h5
            className="categories__item-title"
          >
            Ремонт
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Навчання"
            className="categories__item-img"
            src="./img/book.svg"
          />
          <h5
            className="categories__item-title"
          >
            Навчання
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="IT-послуги"
            className="categories__item-img"
            src="./img/it.svg"
          />
          <h5
            className="categories__item-title"
          >
            IT-послуги
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Фотосесії"
            className="categories__item-img"
            src="./img/camera.svg"
          />
          <h5
            className="categories__item-title"
          >
            Фотосесії
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Прибирання"
            className="categories__item-img"
            src="./img/cleaning.svg"
          />
          <h5
            className="categories__item-title"
          >
            Прибирання
          </h5>
        </a>
      </div>
      <div
        className="categories__item col-sm-6 col-12"
      >
        <a
          href="#"
        >
          <img
            alt="Інше"
            className="categories__item-img"
            src="./img/other.svg"
          />
          <h5
            className="categories__item-title"
          >
            Інше
          </h5>
        </a>
      </div>
    </div>
    <div
      className="categories__link-wrapper"
    >
      <a
        className="categories__all-link link-success"
        href="#"
      >
        Показати всі категорії
        <span
          id="categories__all-link--arrow"
        >
          →
        </span>
      </a>
    </div>
  </section>,
  <footer
    className="footer container-fluid"
    style={
      Object {
        "background": "linear-gradient(
      179deg,
      transparent 0,
      transparent 45px,
      #fff 45px,
      #fff 100%
    )",
      }
    }
  >
    <div
      className="footer__wrapper row row-cols-md-3 row-cols-1"
    >
      <div
        className="footer__column col-md-12 order-md-last"
      >
        <h2
          className="footer__logo"
        >
          <a
            href="/"
          >
            Workers
          </a>
        </h2>
        <p
          className="footer__copy"
        >
          © 2021 workers.com.ua — всі права захищені
        </p>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Реклама на сайті
        </a>
        <a
          href="#"
        >
          Зв'язатися з адміністрацією
        </a>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Угода користувача
        </a>
        <a
          href="#"
        >
          Допомога
        </a>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Замовники
        </a>
        <a
          href="#"
        >
          Виконавці
        </a>
        <a
          href="#"
        >
          Блог
        </a>
        <a
          href="#"
        >
          Безпечна угода
        </a>
      </div>
    </div>
  </footer>,
]
`;
